store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedGetTraitsHandler.java

```diff
@@ -45,35 +37,15 @@
  * return [b]
  */
 public class FederatedGetTraitsHandler implements OutputOperationHandler<GetTraits, Set<StoreTrait>> {
-
     @Override
-    public Set<StoreTrait> doOperation(final GetTraits operation, final Context context, final Store store)
-            throws OperationException {
+    public Set<StoreTrait> doOperation(final GetTraits operation, final Context context, final Store store) throws OperationException {
         try {
-            final FederatedOperationChain<Void, StoreTrait> wrappedFedChain = new FederatedOperationChain.Builder<Void, StoreTrait>()
-                    .operationChain(OperationChain.wrap(operation))
-                    // deep copy options
-                    .options(isNull(operation.getOptions()) ? new HashMap<>() : new HashMap<>(operation.getOptions()))
-                    .build();
-
-            final Iterable<StoreTrait> concatResults = store.execute(wrappedFedChain, context);
-
-            Map<StoreTrait, Integer> rtn;
-            if (nonNull(concatResults) && nonNull(concatResults.iterator()) && concatResults.iterator().hasNext()) {
-                rtn = Streams.toStream(concatResults)
-                        // collect a map of k=trait v=count to covert concat of traits to an intersection of traits.
-                        .collect(Collectors.toMap(t -> t, ignore -> 1,
-                                (existing, replacement) -> existing + replacement));
-
-                final long graphIdsSize = ((FederatedStore) store)
-                        .getGraphs(context.getUser(), operation.getOption(KEY_OPERATION_OPTIONS_GRAPH_IDS), operation)
-                        .stream().count();
-                rtn.values().removeIf(v -> v < graphIdsSize);
-            } else {
-                rtn = Collections.emptyMap();
-            }
-
-            return rtn.keySet();
+            return (Set<StoreTrait>) store.execute(
+                    new FederatedOperation.Builder()
+                            .op(operation)
+                            .mergeFunction(new CollectionIntersect<Object>())
+                            .graphIds(getDeprecatedGraphIds(operation)) // deprecate this line.
+                            .build(), context);
         } catch (final Exception e) {
```

Comment:
Why `Exception`? The thrown message is about traits. Doesn't `FederatedOperation.Builder` throw something more specific than generic `Exception`?

Reply:
It is to catch Any and All exception from execution of the operation not just the builder.
This is to breadcrumb the stack trace for Users and Devs.